docs: restructure dashboard annotations page (#70208)
grafana/grafana: docs/sources/dashboards/build-dashboards/annotate-visualizations/index.md

{{< figure src="/static/img/docs/v46/annotations.png" max-width="800px" >}}

You can annotate visualizations in three ways:

- Directly in the panel, using the [built-in annotations query](#built-in-query)
- Using the HTTP API
- Configuring annotation queries in the dashboard settings

In the first two cases, you're creating new annotations, while in the last you're querying existing annotations from data sources. The built-in query annotation query also supports this.

This page explains the first and third options; for information about using the HTTP API, refer to [Annotations API]({{< relref "../../../developers/http_api/annotations/" >}}).

Annotations are supported for the following visualization types:

- Time series
- State timeline
- Candlestick

## Create annotations in panels

Grafana comes with the ability to add annotation events directly from a panel using the [built-in annotation query](#built-in-query) that exists on all dashboards. Annotations that you create this way are stored in Grafana.

To add annotations directly in the panel, the built-in query must be enabled. Learn more in [Built-in query](#built-in-query)

### Add an annotation

> 

Reviewer: I think our guidelines are to add an intro here like "To add an annotation, complete the following steps:"

Author: I'm doing really light wording edits on this PR because it was primarily to get the restructuring done and some of the language to make any sense at all. I'll do a full sweep of this when I get more nitty gritty on this content.

Reviewer: got it. 👍

1. In the dashboard click the panel to which you're adding the annotation. A context menu will appear.
1. In the context menu, click **Add annotation**.
   ![Add annotation context menu](/static/img/docs/time-series-panel/time-series-annotations-context-menu.png)
1. Add an annotation description and tags (optional).
1. Click **Save**.

> Alternatively, to add an annotation, press Ctrl/Cmd and click the panel, and the **Add annotation** popover will appear

Reviewer: ```suggestion
Alternatively, to add an annotation, press Ctrl/Cmd and click the panel, and the **Add annotation** popover will appear.
```

# Annotate visualizations

Annotations provide a way to mark points on a visualization with rich events. They are visualized as vertical lines and icons on all graph panels. When you hover over an annotation, you can get event description and event tags. The text field can include links to other systems with more detail.

{{< figure src="/static/img/docs/v46/annotations.png" max-width="800px" >}}

You can annotate visualizations in three ways:

- Directly in the panel, using the [built-in annotations query](#built-in-query)
- Using the HTTP API
- Configuring annotation queries in the dashboard settings

> In the first two cases, you're creating new annotations, while in the last you're querying existing annotations from data sources. The built-in query annotation query also supports this.

Author: ```suggestion
In the first two cases, you're creating new annotations, while in the last you're querying existing annotations from data sources. The built-in annotation query also supports this.
```

1. Add an annotation description and tags (optional).
1. Click **Save**.

Alternatively, to add an annotation, press Ctrl/Cmd and click the panel, and the **Add annotation** popover will appear.

### Add a region annotation

1. In the dashboard press Ctrl/Cmd and click and drag on the panel.
   ![Add annotation popover](/static/img/docs/time-series-panel/time-series-annotations-add-region-annotation.gif)
1. Add an annotation description and tags (optional).
1. Click **Save**.

### Edit an annotation

1. In the dashboard, hover over an annotation indicator on the Time series panel.
   <!--![Add annotation popover](/static/img/docs/time-series-panel/time-series-annotations-edit-annotation.gif)-->
1. Click on the pencil icon in the annotation tooltip.
1. Modify the description and/or tags.
1. Click save.

### Delete an annotation

1. In the dashboard hover over an annotation indicator on a panel.
   <!--![Add annotation popover](/static/img/docs/time-series-panel/time-series-annotations-edit-annotation.gif)-->
1. Click on the trash icon in the annotation tooltip.

## Fetch annotations through dashboard settings

> In the dashboard settings, under **Annotations**, you can add new queries to fetch annotations using any data source, including the the built-in data annotation data source. Annotation queries return events that can be visualized as event markers in graphs across the dashboard.

Reviewer 2: ```suggestion
In the dashboard settings, under **Annotations**, you can add new queries to fetch annotations using any data source, including the built-in data annotation data source. Annotation queries return events that can be visualized as event markers in graphs across the dashboard.
```

{{< figure src="/static/img/docs/v46/annotations.png" max-width="800px" >}}

> You can annotate panels in three ways:

Reviewer 2: If " annotate " means creating a new annotation, this is misleading. There are 2 ways to create annotations, from the UI (panels) or the HTTP API. 
If we are talking about fetching annotations, I would say there is only one way: configuring annotations queries (built-in comes by default) in the dashboard settings.

Author: I mean it in the most general sense of the word, that you get annotations onto your panels in different ways. Maybe I could say something like this to clarify:

```suggestion
You can annotate panels in three ways. 

- Directly in the panel, using the [built-in annotations query](#built-in-query)
- Using the HTTP API
- Configure annotation queries in the dashboard settings

In the first two cases, you are creating new annotations, while in the last you're querying existing annotations from data sources. The built-in query annotation query also supports this.
```

- Directly in the panel, using the built-in annotations query (add link)

> - Querying other data sources, using the dashboard settings (add link)

Author: ```suggestion
- Adding annotation queries for any data source, using the dashboard settings (add link)
```

- Using the HTTP API

This page explains the first two options; for information about using the HTTP API, refer to [Annotations API]({{< relref "../../../developers/http_api/annotations/" >}}).

> ## Add annotations in panels

Author: ```suggestion
## Create annotations in panels
```

   ![Add annotation context menu](/static/img/docs/time-series-panel/time-series-annotations-context-menu.png)
1. Add an annotation description and tags(optional).
1. Click save.

Alternatively, to add an annotation, Ctrl/Cmd+Click on the Time series panel and the Add annotation popover will appear

### Add a region annotation

1. In the dashboard Ctrl/Cmd+click and drag on the Time series panel.
   ![Add annotation popover](/static/img/docs/time-series-panel/time-series-annotations-add-region-annotation.gif)
1. Add an annotation description and tags(optional).
1. Click save.

### Edit an annotation

1. In the dashboard hover over an annotation indicator on the Time series panel.
   <!--![Add annotation popover](/static/img/docs/time-series-panel/time-series-annotations-edit-annotation.gif)-->
1. Click on the pencil icon in the annotation tooltip.
1. Modify the description and/or tags.
1. Click save.

### Delete an annotation

1. In the dashboard hover over an annotation indicator on the Time series panel.
   <!--![Add annotation popover](/static/img/docs/time-series-panel/time-series-annotations-edit-annotation.gif)-->
1. Click on the trash icon in the annotation tooltip.

> ## Add annotations through dashboard settings

Author: ```suggestion
## Add annotations queries in the dashboard settings
```

Reviewer 2: In other places we mention the **Annotation settings**, which are accessible through **Dashboard settings**.
I would like to distinguish between **Create** and **Fetch**. Sometimes we refer to **Add** to create new annotations and sometimes we use it to **Fetch** existent annotations from other places. Not sure how to do it though

Author: What about this? I've also changed the previous major heading to "Create annotations in the panel" to distinguish the two actions.
```suggestion
## Fetch annotations through dashboard settings
```

> In the dashboard settings you can create new queries to fetch annotations from the native annotation store using the built-in data annotation data source or add query other data sources.

Reviewer 2: Should we say Annotation settings and then explain how to get there?

Reviewer 2: I found a nice starting explanation inside Grafana, and there is no mention about **Native** annotations. Should we still name them like this? I would be great to be consistent between this doc and Grafana Docs.
![image](https://github.com/grafana/grafana/assets/20256983/24dade50-c06b-4d75-adfe-38ee1d8352ae)

Author: How to get to the annotation settings is explained in 73-75. And yeah, that explanation is actually really good. I'll add some of that into the introduction.

> In the dashboard settings you can create new queries to fetch annotations from the native annotation store using the built-in data annotation data source or add query other data sources.

Author: Removing that "native" language
```suggestion
In the dashboard settings, under **Annotations**, you can add new queries to fetch annotations using the built-in data annotation data source or other data sources.
```

After you add an annotation they will still be visible. This is due to the built in annotation query that exists on all dashboards. This annotation query will fetch all annotation events that originate from the current dashboard and show them on the panel where they were created. This includes alert state history annotations. You can stop annotations from being fetched and drawn by opening the **Annotations** settings (via Dashboard cogs menu) and modifying the query named `Annotations & Alerts (Built-in)`.

When you copy a dashboard using the **Save As** feature it will get a new dashboard id so annotations created on source dashboard will no longer be visible on the copy. You
can still show them if you add a new **Annotation Query** and filter by tags. But this only works if the annotations on the source dashboard had tags to filter by.

If this query isn't enabled, you can't add annotations directly in dashboard panels.

Following are some query options specific to the built-in annotation query.

### Filter queries by tag

You can create new queries to fetch annotations from the native annotation store via the `-- Grafana --` data source by setting _Filter by_ to `Tags`.

> Grafana v8.1 and later versions also support typeahead of existing tags, provide at least one tag.

Reviewer 2: Do we still need to mention that this specific feature started at v8.1?

Author: I feel like we're still figuring out the guidance around these notes. This one isn't super old, so I'm gonna leave it in.

For example, create an annotation query name `outages` and specify a tag `outage`. This query will show all annotations (from any dashboard or via API) with the `outage` tag. If multiple tags are defined in an annotation query, then Grafana will only show annotations matching all the tags. To modify the behavior, enable `Match any`, and Grafana will show annotations that contain any one of the tags you provided.

{{< figure src="/media/docs/grafana/dashboards/screenshot-annotations-typeahead-support-10.0.png" max-width="600px" >}}

> In Grafana v5.3+ it's possible to use template variables in the tag query. So if you have a dashboard showing stats for different services and a template variable that dictates which services to show, you can now use the same template variable in your annotation query to only show annotations for those services.

Reviewer 2: Same here, I don't know what the convention should be but do we need to mention v5.3?

Author: Agreed, this reference seems pretty old to have in here.
```suggestion
You can also use template variables in the tag query. This means if you have a dashboard showing stats for different services and a template variable that dictates which services to show, you can use the same template variable in your annotation query to only show annotations for those services.
```
